src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -767,15 +767,18 @@ await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '.')
                 int status = 100 * (status1 - '0') + 10 * (status2 - '0') + (status3 - '0');
                 response.StatusCode = (HttpStatusCode)status;
 
-                if (await ReadCharAsync(cancellationToken).ConfigureAwait(false) != ' ')
+                // Parse (optional) reason phrase
+                _sb.Clear();
+                char c = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
+                if (c == ' ')
+                {
+                    c = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else if (c != '\r')
```

Comment:
The RFC doesn't seem to allow this, but if we allow it today, I suppose there's no harm here.